text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+                "componentClasses": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/Class"
+                    }
+                },
+                "imports": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Import"
+                    }
+                },
+                "dynamicImports": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/DynamicImport"
+                    }
+                },
+                "exports": {
+                    "type": "array",
+                    "items": [
+                        { "$ref": "#/definitions/Export"},
+                        { "$ref": "#/definitions/AggregatingExport"}
+                    ]
+                },
+                "programmaticEvents": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ProgrammaticEvent"
+                    }
+                },
+                "staticResources": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/StaticResourceReference"
+                    }
+                }
+            }
+        },
+        "CSSFile": {
+            "type": "object",
+            "properties": {
+                "fileType": { "$ref": "#/definitions/FileType" },
+                "fileName": { "type": "string" },
+                "tokens": { "type": "array", "items": { "type": "string" }},
+                "staticResources": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/StaticResourceReference"
+                    }
+                }
+            }
+        },
+        "AggregatingExport": {
+            "type": "object",
+            "properties": {
+                "exportsList": { 
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "enum": ["class", "function", "expression"]
+                            }
+                        }
+                    }
+                },
+                "moduleSpecifier": { "$ref": "#/definitions/ModuleReference" }
+            }
+        },
+        "ApiDecorator": {
+            "type": "object",
+            "properties": {
+                "type": { "const": "api" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "Class": {
+            "type": "object",
+            "properties": {
+                "name": { "type": "string" },
+                "extends": { "$ref": "#/definitions/Class" },
```

Comment:
Just to confirm, the `extends` value will be present if the base class is defined in the file file. Right?

Reply:
It should accommodate external class references as well. I have added a new ParentClass definition to accommodate imported classes.